packages/stream-metadata/src/routes/spaceMetadata.ts

```diff
@@ -1,36 +1,89 @@
-import { FastifyRequest, FastifyReply } from 'fastify'
+import { FastifyReply, FastifyRequest } from 'fastify'
+import { SpaceInfo } from '@river-build/web3'
+import { z } from 'zod'
 
+import { config } from '../environment'
 import { isValidEthereumAddress } from '../validators'
+import { spaceDapp } from '../contract-utils'
 
-export function fetchSpaceMetadata(
-	request: FastifyRequest,
-	reply: FastifyReply,
-	serverUrl: string,
-) {
+export const spaceMetadataBaseUrl = `${config.streamMetadataBaseUrl}space`.toLowerCase()
+
+const paramsSchema = z.object({
+	spaceAddress: z.string().min(1, 'spaceAddress parameter is required'),
+})
+
+export interface SpaceMetadataResponse {
+	name: string
+	description: string
+	image: string
+}
+
+export async function fetchSpaceMetadata(request: FastifyRequest, reply: FastifyReply) {
 	const logger = request.log.child({ name: fetchSpaceMetadata.name })
-	const { spaceAddress } = request.params as { spaceAddress?: string }
-	logger.info({ spaceAddress }, 'GET /space/../metadata')
 
-	if (!spaceAddress) {
-		return reply
-			.code(400)
-			.send({ error: 'Bad Request', message: 'spaceAddress parameter is required' })
+	const parseResult = paramsSchema.safeParse(request.params)
+
+	if (!parseResult.success) {
+		const errorMessage = parseResult.error.errors[0]?.message || 'Invalid parameters'
+		logger.info(errorMessage)
+		return reply.code(400).send({ error: 'Bad Request', message: errorMessage })
 	}
 
+	const { spaceAddress } = parseResult.data
+
 	// Validate spaceAddress format using the helper function
 	if (!isValidEthereumAddress(spaceAddress)) {
+		logger.info({ spaceAddress }, 'Invalid spaceAddress format')
 		return reply
 			.code(400)
 			.send({ error: 'Bad Request', message: 'Invalid spaceAddress format' })
 	}
 
-	const dummyJson = {
-		name: '....',
-		description: '....',
-		members: 99999,
-		fees: '0.001 eth',
-		image: `${serverUrl}/space/${spaceAddress}/image`,
+	let spaceInfo: SpaceInfo | undefined
+	try {
+		spaceInfo = await spaceDapp.getSpaceInfo(spaceAddress)
+	} catch (error) {
+		logger.error({ spaceAddress, error }, 'Failed to fetch space contract info')
+		return reply
+			.code(404)
+			.send({ error: 'Not Found', message: 'Failed to fetch space contract info' })
+	}
```

Comment:
i don't like that this will eat any error from infura and return a 404 instead of 500 if infura is down